src/services/Parse.js

```diff
@@ -0,0 +1,118 @@
+Parse.serverURL = PARSE_HOST_URL;
+
+export default Parse;
+
+export const createEvent = async (eventData, userId) => {
+    try {
+        const Event = Parse.Object.extend("Event");
+        const newEvent = new Event();
+
+        newEvent.set("heading", eventData.headline); // Event title
+        newEvent.set("description", eventData.description); // Event description
+        newEvent.set("datetime", eventData.datetime); // Event date and time
+        newEvent.set("location", eventData.location); // Event location 
+        newEvent.set("petType", eventData.eventPetType); // Type of pet related to the event (e.g., Dog, Cat, Bird)
+
+        // Set the image URL or file path
+        newEvent.set("image", eventData.image || ""); // Optional, default to empty if not provided
+
+        // Set the created_by field with a reference to the User
+        const User = Parse.Object.extend("User");
+        const userPointer = new User();
+        userPointer.id = userId; // Assuming userId is passed as an argument
+        newEvent.set("created_by", userPointer); // Foreign key reference
+
+        await newEvent.save();
+        console.log("Event created successfully!");
+    } catch (error) {
+        console.error("Error while creating event:", error);
+    }
+};
+
+
+export const fetchEventDetails = async (eventId) => {
+    try {
+        const Event = Parse.Object.extend("Event");
+        const query = new Parse.Query(Event);
+
+        query.equalTo("objectId", eventId); // Search by event ID
+        const event = await query.first(); // Fetch the first matching result
+
+        if (event) {
+            // Return the event details as an object
+            return {
+                id: event.id,
+                headline: event.get("heading"), // Event title
+                description: event.get("description"), // Event description
+                datetime: event.get("datetime"), // Event date and time
+                location: event.get("location"), // Event location
+                petType: event.get("petType"), // Type of pet
+                image: event.get("image"), // Image URL or path
+                createdBy: event.get("created_by").id, // User ID who created the event (pointer to User)
+            };
+        } else {
+            console.log("No event found with this ID");
+            return null;
+        }
+    } catch (error) {
+        console.error("Error fetching event details:", error);
+        return null;
+    }
+};
+
+
+
+
+// 检查用户是否对某个事件感兴趣
+export const checkUserInterest = async (eventId, userId) => {
+    try {
+        const EventParticipation = Parse.Object.extend("EventParticipation");
+        const query = new Parse.Query(EventParticipation);
+
+        query.equalTo("event_id", eventId);
+        query.equalTo("user_id", userId);
+        const existingRecord = await query.first();
+
+        return !!existingRecord; // 返回布尔值，表示是否存在记录
```

Comment:
This line effectively converts the existence check to a boolean, which is great. Ensure that any false positives are handled upstream if needed. Also, be sure to understand how it works ;)